[synthetics] Add docs on private locations (#2075)
elastic/observability-docs: docs/en/observability/uptime-set-up-heartbeat.asciidoc

[[uptime-set-up-choose-heartbeat]]
= Use {heartbeat}

> {heartbeat} is a lightweight daemon that you install on a remote server to periodically
> check the status of your services and determine if they are available. Performance data is
> then gathered and reported back to {heartbeat}, where it's formatted and sent to the {stack}.

Reviewer: Sorry for the slightly pedantic suggestion, but just to be precise here I'd say that Heartbeat collects performance data rather than having it reported to it (unless I'm missing something?):

```suggestion
{heartbeat} is a lightweight daemon that you install on a remote server to periodically
check the status of your services and determine if they are available. It gathers performance data,
formats it, and sends the data to the {stack}.
```

[discrete]
[[uptime-set-up-docker]]
== Pull the Docker image

> Elastic provides Docker images that you can use to run monitors, which reduces the chance of problems.

Reviewer: I'd say it makes it "easier to run", as mentioned previously. Besides not having to advocate too strongly for Docker, IMO saying it "reduces the chance of problems" reflects badly on us (as in: "ah so there are significant chances of problems")

Start by pulling the {heartbeat} Docker image.

// ifeval::["{release-state}"=="unreleased"]

// Version {version} has not yet been released.

// endif::[]

// ifeval::["{release-state}"!="unreleased"]

[source,sh,subs="attributes"]
----
docker pull docker.elastic.co/beats/heartbeat:{version}
----

// endif::[]

[discrete]
[[uptime-set-up-config]]
== Configure

Next, create a `heartbeat.yml` configuration file.

> There are four possible monitor types.

Reviewer: Do we need to say there are four types if we're going to explain HTTP straightaway? IMO we should either list what the types are or (preferably IMO) remove this phrase.

It seems a bit out of context here.

The example below shows how to configure an `http` monitor.

[source,yml]
----
heartbeat.monitors:
- type: http
  id: service-status <1>
  name: Service Status
  service.name: my-apm-service-name
  hosts: ["http://localhost:80/service/status"]
  check.response.status: [200]
  schedule: '@every 5s'
----
<1> Each `monitor` gets its own ID in the {uptime-app} and its own schedule entry.
This allows tests to be run in parallel and analyzed separately.

Read more about configuration options in {heartbeat-ref}/configuration-heartbeat-options.html[Configure {heartbeat} monitors].

If you are setting up a **`browser`** monitor, see <<synthetics-create-test>>

> for more details on writing synthetic tests to be added to the configuration file.

Reviewer: This phrase sounds confusing, especially the last part about "for more details on writing synthetic tests to be added to the configuration file".

I don't understand what this means, so can't provide suggestions, but does sound a bit weird IMO 😢

Author: Is this clearer? 

```suggestion
NOTE: If you are setting up a **`browser`** monitor, see <<synthetics-create-test>>
for more details on writing synthetic tests. You can add tests directly to the configuration file 
using https://www.elastic.co/guide/en/beats/heartbeat/current/monitor-browser-options.html#monitor-source-inline[`inline`] or reference tests stored elsewhere using https://www.elastic.co/guide/en/beats/heartbeat/current/monitor-browser-options.html#monitor-source-zipurl[`zip_url`].
```

Reviewer 2: I like the rewording, but I suppose I didn't find the original confusing myself.

Reviewer: I like the rewording, just one nit:

Instead of:

```
You can add tests directly to the configuration file [...]
```

Maybe we should say:

```
You can add tests directly when configuring the monitor through the UI [...] 
```
